internal/testsacc/alb_virtual_service_resource_test.go

```diff
@@ -0,0 +1,224 @@
+	resp.Append(GetDataSourceConfig()[EdgeGatewayDataSourceName]().GetSpecificConfig("example_with_id"))
+	resp.Append(GetResourceConfig()[ALBPoolResourceName]().GetDefaultConfig)
+	// resp.Append(GetDataSourceConfig()[PublicIPResourceName]().GetDefaultConfig)
+	return
+}
+
+func (r *ALBVirtualServiceResource) Tests(ctx context.Context) map[testsacc.TestName]func(ctx context.Context, resourceName string) testsacc.Test {
+	return map[testsacc.TestName]func(ctx context.Context, resourceName string) testsacc.Test{
+		// TODO : Complete tests
+		// * First with service_type: HTTP and Simple Ports
+		"example": func(_ context.Context, resourceName string) testsacc.Test {
+			return testsacc.Test{
+				CommonChecks: []resource.TestCheckFunc{
+					resource.TestCheckResourceAttrSet(resourceName, "id"),
+					resource.TestCheckResourceAttrSet(resourceName, "name"),
+					resource.TestCheckResourceAttrSet(resourceName, "pool_id"),
+				},
+				// ! Create testing with service_type: HTTP
+				Create: testsacc.TFConfig{
+					TFConfig: testsacc.GenerateFromTemplate(resourceName, `
+					resource "cloudavenue_alb_virtual_service" "example" {
+						name = {{ generate . "name" }}
+						description = {{ generate . "description" }}
+						edge_gateway_id = data.cloudavenue_edgegateway.example_with_id.id
+						enabled = true
+						pool_id = cloudavenue_alb_pool.example.id
+						virtual_ip = "192.168.10.10"
+						service_type = "HTTP"
+						service_ports = [
+							{
+								port_start = 80
+							}
+						]
+					}`),
+					Checks: []resource.TestCheckFunc{
+						resource.TestCheckResourceAttr(resourceName, "name", testsacc.GetValueFromTemplate(resourceName, "name")),
+						resource.TestCheckResourceAttr(resourceName, "description", testsacc.GetValueFromTemplate(resourceName, "description")),
+						resource.TestCheckResourceAttrSet(resourceName, "edge_gateway_id"),
+						resource.TestCheckResourceAttr(resourceName, "enabled", "true"),
+						resource.TestCheckResourceAttr(resourceName, "virtual_ip", "192.168.10.10"),
+						resource.TestCheckResourceAttr(resourceName, "service_type", "HTTP"),
+						resource.TestCheckTypeSetElemNestedAttrs(resourceName, "service_ports.*", map[string]string{
+							"port_start": "80",
+						}),
+					},
+				},
+				// ! Updates testing with service_type: HTTPS
+				// * Add a new port 443 with SSL and another port 8443 without SSL with service_type: HTTPS
+				Updates: []testsacc.TFConfig{
+					{
+						TFConfig: testsacc.GenerateFromTemplate(resourceName, `
+							resource "cloudavenue_alb_virtual_service" "example" {
+								name = {{ get . "name" }}
+								description = {{ generate . "description" }}
+								edge_gateway_id = data.cloudavenue_edgegateway.example_with_id.id
+								enabled = true
+								pool_id = cloudavenue_alb_pool.example.id
+								virtual_ip = "192.168.10.10"
+								certificate_id = "urn:vcloud:certificateLibraryItem:f9caac3a-2555-477e-ae58-0740687d4daf"
```

Comment:
Add comment to indicate that this is temporary and must be replaced by the resource or datasource once it is created